src/platform_impl/linux/kms/event_loop.rs

```diff
@@ -0,0 +1,701 @@
+use parking_lot::Mutex;
+use raw_window_handle::{DrmDisplayHandle, RawDisplayHandle};
+use std::{
+    cell::RefCell,
+    collections::VecDeque,
+    marker::PhantomData,
+    os::fd::AsRawFd,
+    path::{Path, PathBuf},
+    rc::Rc,
+    sync::{mpsc::SendError, Arc},
+    time::{Duration, Instant},
+};
+#[cfg(feature = "kms-ext")]
+use std::{collections::HashMap, sync::atomic::AtomicBool};
+use udev::Enumerator;
+use xkbcommon::xkb;
+
+use drm::control::*;
+
+use crate::error;
+
+use crate::{
+    dpi::PhysicalPosition,
+    event::{Event, StartCause},
+    event_loop::{self, ControlFlow, EventLoopClosed},
+    platform_impl::{self, platform::sticky_exit_callback, MonitorHandle, OsError},
+};
+
+use super::{
+    input::{Interface, LibinputInputBackend},
+    window::winid,
+    Card, MODE,
+};
+
+/// An event loop's sink to deliver events from the Wayland event callbacks
+/// to the winit's user.
+pub type EventSink = Vec<Event<'static, ()>>;
+
+pub struct EventLoopWindowTarget<T> {
+    /// Drm Connector
+    pub connector: connector::Info,
+
+    /// Drm crtc
+    pub crtc: crtc::Info,
+
+    /// Drm plane
+    pub plane: plane::Handle,
+
+    /// Allows window to edit cursor position
+    pub(crate) cursor_arc: Arc<Mutex<PhysicalPosition<f64>>>,
+
+    /// Drm device
+    pub device: Card,
+
+    /// Event loop handle.
+    pub event_loop_handle: calloop::LoopHandle<'static, EventSink>,
+
+    pub(crate) event_sink: EventSink,
+
+    /// A proxy to wake up event loop.
+    pub event_loop_awakener: calloop::ping::Ping,
+
+    _marker: std::marker::PhantomData<T>,
+}
```

Comment:
It would be nice if all of the fields were either `pub` or `pub(crate)`, instead of this mix.